bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/entity/pcm/SEFFActionSequenceElement.java

```diff
@@ -94,6 +106,11 @@ public String toString() {
     		if (seff.isPresent()) {
     			elementName = "Beginning " + seff.get().getDescribedService__SEFF().getEntityName();
     		}
+    		if (this.isBranching()) {
+    			Optional<BranchAction> branchAction = PCMQueryUtils.findParentOfType(this.getElement(), BranchAction.class, false);
+    			Optional<AbstractBranchTransition> branchTransition = PCMQueryUtils.findParentOfType(this.getElement(), AbstractBranchTransition.class, false);
+    			elementName = "Branching " + seff.get().getDescribedService__SEFF().getEntityName() + "." + branchAction.get().getEntityName() + "." +  branchTransition.get().getEntityName();
```

Comment:
I wonder if this naming scheme still correctly represents the structure in nested branches, i.e., branches in branches.

Reply:
It should find the innermost branch that the element (i.e. Start Action) is contained in